application/models/Permission.php

```diff
@@ -720,15 +724,35 @@ private static function comparePermissionTitle($aApermission, $aBpermission)
     public function getUserId($iUserID = null)
     {
         if (empty($iUserID)) {
-            if (Yii::app() instanceof CConsoleApplication) {
-                /* Alt : return 1st forcedAdmin ? */
+            if (App() instanceof CConsoleApplication) {
+                /* Alt : return 1st forcedSuperAdmin ? */
                 throw new Exception('Permission must not be tested with console application.');
             }
-            /* See TestBaseClass tearDownAfterClass */
-            $iUserID = Yii::app()->session['loginID'];
+            return self::getCurrentUserId();
         }
         return $iUserID;
     }
+
+    /**
+     * get the current id of connected user
+     * @return int|null user id
+     * @throws Exception
+     */
+    public static function getCurrentUserId()
```

Comment:
Maybe in https://github.com/LimeSurvey/LimeSurvey/blob/master/application/core/LSWebUser.php ?

Reply:
Why not in LsYiiApplication?

Reply:
Yes, I think this has to be moved.